instant-xml/src/lib.rs

```diff
@@ -10,15 +10,47 @@ pub use macros::{FromXml, ToXml};
 pub mod parse;
 
 pub trait ToXml {
-    fn write_xml<W: fmt::Write>(&self, write: &mut W) -> Result<(), Error>;
+    fn write_xml<W: fmt::Write>(
+        &self,
+        write: &mut W,
+        parent_prefixes: Option<&mut BTreeSet<&str>>,
+    ) -> Result<(), Error>;
 
-    fn to_xml(&self) -> Result<String, Error> {
+    fn to_xml(&self, parent_prefixes: Option<&mut BTreeSet<&str>>) -> Result<String, Error> {
         let mut out = String::new();
-        self.write_xml(&mut out)?;
+        self.write_xml(&mut out, parent_prefixes)?;
         Ok(out)
     }
 }
 
+macro_rules! to_xml_for_type {
+    ($typ:ty) => {
+        impl ToXml for $typ {
+            fn write_xml<W: fmt::Write>(
+                &self,
+                _write: &mut W,
+                _parent_prefixes: Option<&mut BTreeSet<&str>>,
+            ) -> Result<(), Error> {
+                Ok(())
+            }
+
+            fn to_xml(
+                &self,
+                parent_prefixes: Option<&mut BTreeSet<&str>>,
+            ) -> Result<String, Error> {
+                self.write_xml(&mut self.to_string(), parent_prefixes)?;
+                Ok(self.to_string())
+            }
```

Comment:
We don't need to provide `to_xml()` here, since the trait provides a default implementation.